### PR TITLE
New PDS version + refactor naming in helm and docker parts

### LINKED ISSUE #686: Enhance PDS gosec docker and helm parts
## Env file
Please rename `env` to `env-initial` and use this file to copy into missing `.env`. So it's clear that this is the intiial part.

## Structure
At 
https://github.com/Daimler/sechub/tree/develop/sechub-pds-examples/gosec

I see
```
context
pds-gosec
...
pds_gosec_alpine.yaml
```
It was a little bit confusing what part is exactly for which purpose.

I would prefer to have
```
docker (instead context)
docker-compose_pds_gosec_alpine.yaml  - we are not allowed to have subfolder "docker-compose" for security reasons, but using this name we would made clear for which this file is exactly: docker compose...
docker-compose_pds_gosec_ubuntu.yaml
helm/pds-gosec
```
## Naming of docker compose start
When having `helm` folder it becomes clear that this is for k8s, but that "01-start-local.sh" does NOT start the server simply as a jar, but doing a docker-compose is not clear automatically.

Please rename to `01-start-local-docker-compose.sh`

### PR DESCRIPTION
**Pull request by Jeeppler**
- Add new PDS version 0.23.1 which can run in cluster #671.
- Local cluster shared storage and S3 using MinIO
- Refactor the structure and files based on previous feedback from @de-jcup 
- Improve Helm chart to work with shared storage and S3 #686 
- Restructure, refactor and improve the README.adoc
- Include pictures into the README.adoc

--

- closes: #671
- closes #686

### CODE REVIEW

**Comment by de-jcup** [comment]
Please describe usage of `70-test.sh` a little bit - at least an info that this script is needing an existing PDS server or cluster running etc.

File: sechub-pds-examples/gosec/50-start-multiple-docker-compose.sh [hunk 1]
```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env bash
+# SPDX-License-Identifier: MIT
+
+SCRIPT_DIR=`dirname $0`
+REPLICAS="$1"
+
+ENVIRONMENT_FILE=".env-cluster"
+
```
Comment (de-jcup) [inline sechub-pds-examples/gosec/50-start-multiple-docker-compose.sh:8]: I started with `./50-start-multiple-docker-compose.sh 3`

after this I tried out with sechub to connect.

The problem here: SecHub adapter does not accept older TLS versions - the load balancer implementation seems to use old ones.

SecHub scanning mit single variant (01-... script) does work!

```
2021-09-02 14:43:53.711 ERROR 28739 --- [-7fbfd54100fb-2] .s.d.s.p.AbstractProductExecutionService : Product executor failed:PDS_CODESCAN __[SECHUB-UID:e374fa6e-bd5c-45ff-953e-7fbfd54100fb]__

com.daimler.sechub.sharedkernel.execution.SecHubExecutionException: PDS_CODESCAN execution failed.__[SECHUB-UID:e374fa6e-bd5c-45ff-953e-7fbfd54100fb]__
	at com.daimler.sechub.domain.scan.product.AbstractInstallSetupProductExecutor.execute(AbstractInstallSetupProductExecutor.java:85)
	at com.daimler.sechub.domain.scan.product.AbstractProductExecutionService.execute(AbstractProductExecutionService.java:93)
	at com.daimler.sechub.domain.scan.product.AbstractProductExecutionService.runOnExecutorWithOneConfiguration(AbstractProductExecutionService.java:187)
	at com.daimler.sechub.domain.scan.product.AbstractProductExecutionService.runOnAllAvailableExecutors(AbstractProductExecutionService.java:151)
	at com.daimler.sechub.domain.scan.product.AbstractProductExecutionService.executeProductsAndStoreResults(AbstractProductExecutionService.java:68)
	at com.daimler.sechub.domain.scan.ScanJobExecutor$CanceableScanJobRunnable.run(ScanJobExecutor.java:145)
	at java.lang.Thread.run(Thread.java:748)
Caused by: org.springframework.web.client.ResourceAccessException: I/O error on POST request for "https://localhost:8444/api/job/create": Received fatal alert: protocol_version; nested exception is javax.net.ssl.SSLHandshakeException: Received fatal alert: protocol_version
	at org.springframework.web.client.RestTemplate.doExecute(RestTemplate.java:785)
	at org.springframework.web.client.RestTemplate.execute(RestTemplate.java:711)
	at org.springframework.web.client.RestTemplate.postForEntity(RestTemplate.java:468)
	at com.daimler.sechub.adapter.support.RestOperationsSupport.postJSON(RestOperationsSupport.java:28)
	at com.daimler.sechub.adapter.pds.PDSAdapterV1.createNewPDSJOB(PDSAdapterV1.java:193)
	at com.daimler.sechub.adapter.pds.PDSAdapterV1.execute(PDSAdapterV1.java:46)
	at com.daimler.sechub.adapter.pds.DelegatingMockablePDSAdapterV1.execute(DelegatingMockablePDSAdapterV1.java:53)
	at com.daimler.sechub.adapter.pds.DelegatingMockablePDSAdapterV1.execute(DelegatingMockablePDSAdapterV1.java:1)
	at com.daimler.sechub.adapter.AbstractAdapter.start(AbstractAdapter.java:71)
	at com.daimler.sechub.domain.scan.product.pds.PDSCodeScanProductExecutor.lambda$0(PDSCodeScanProductExecutor.java:130)
	at com.daimler.sechub.sharedkernel.resilience.ResilientActionExecutor.executeResilient(ResilientActionExecutor.java:53)
	at com.daimler.sechub.sharedkernel.resilience.ResilientActionExecutor.executeResilient(ResilientActionExecutor.java:39)
	at com.daimler.sechub.domain.scan.product.pds.PDSCodeScanProductExecutor.executeWithAdapter(PDSCodeScanProductExecutor.java:94)
	at com.daimler.sechub.domain.scan.product.pds.PDSCodeScanProductExecutor.executeWithAdapter(PDSCodeScanProductExecutor.java:1)
	at com.daimler.sechub.domain.scan.product.AbstractInstallSetupProductExecutor.executeAdapterWhenTargetTypeSupported(AbstractInstallSetupProductExecutor.java:191)
	at com.daimler.sechub.domain.scan.product.AbstractInstallSetupProductExecutor.execute(AbstractInstallSetupProductExecutor.java:165)
	at com.daimler.sechub.domain.scan.product.AbstractInstallSetupProductExecutor.execute(AbstractInstallSetupProductExecutor.java:77)
	... 6 common frames omitted
Caused by: javax.net.ssl.SSLHandshakeException: Received fatal alert: protocol_version
	at sun.security.ssl.Alert.createSSLException(Alert.java:131)
	at sun.security.ssl.Alert.createSSLException(Alert.java:117)
	at sun.security.ssl.TransportContext.fatal(TransportContext.java:311)
	at sun.security.ssl.Alert$AlertConsumer.consume(Alert.java:293)
	at sun.security.ssl.TransportContext.dispatch(TransportContext.java:185)
	at sun.security.ssl.SSLTransport.decode(SSLTransport.java:149)
	at sun.security.ssl.SSLSocketImpl.decode(SSLSocketImpl.java:1143)
	at sun.security.ssl.SSLSocketImpl.readHandshakeRecord(SSLSocketImpl.java:1054)
	at sun.security.ssl.SSLSocketImpl.startHandshake(SSLSocketImpl.java:394)
	at org.apache.http.conn.ssl.SSLConnectionSocketFactory.createLayeredSocket(SSLConnectionSocketFactory.java:436)
	at org.apache.http.conn.ssl.SSLConnectionSocketFactory.connectSocket(SSLConnectionSocketFactory.java:384)
	at org.apache.http.impl.conn.DefaultHttpClientConnectionOperator.connect(DefaultHttpClientConnectionOperator.java:142)
	at org.apache.http.impl.conn.PoolingHttpClientConnectionManager.connect(PoolingHttpClientConnectionManager.java:374)
	at org.apache.http.impl.execchain.MainClientExec.establishRoute(MainClientExec.java:393)
	at org.apache.http.impl.execchain.MainClientExec.execute(MainClientExec.java:236)
	at org.apache.http.impl.execchain.ProtocolExec.execute(ProtocolExec.java:186)
	at org.apache.http.impl.execchain.RetryExec.execute(RetryExec.java:89)
	at org.apache.http.impl.execchain.RedirectExec.execute(RedirectExec.java:110)
	at org.apache.http.impl.client.InternalHttpClient.doExecute(InternalHttpClient.java:185)
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:83)
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:56)
	at org.springframework.http.client.HttpComponentsClientHttpRequest.executeInternal(HttpComponentsClientHttpRequest.java:87)
	at org.springframework.http.client.AbstractBufferingClientHttpRequest.executeInternal(AbstractBufferingClientHttpRequest.java:48)
	at org.springframework.http.client.AbstractClientHttpRequest.execute(AbstractClientHttpRequest.java:66)
	at org.springframework.http.client.BufferingClientHttpRequestWrapper.executeInternal(BufferingClientHttpRequestWrapper.java:63)
	at org.springframework.http.client.AbstractBufferingClientHttpRequest.executeInternal(AbstractBufferingClientHttpRequest.java:48)
	at org.springframework.http.client.AbstractClientHttpRequest.execute(AbstractClientHttpRequest.java:66)
	at org.springframework.http.client.InterceptingClientHttpRequest$InterceptingRequestExecution.execute(InterceptingClientHttpRequest.java:109)
	at com.daimler.sechub.adapter.pds.PDSClientHttpRequestInterceptor.intercept(PDSClientHttpRequestInterceptor.java:36)
	at org.springframework.http.client.InterceptingClientHttpRequest$InterceptingRequestExecution.execute(InterceptingClientHttpRequest.java:93)
	at com.daimler.sechub.adapter.TraceLogClientHTTPRequestInterceptor.intercept(TraceLogClientHTTPRequestInterceptor.java:32)
	at org.springframework.http.client.InterceptingClientHttpRequest$InterceptingRequestExecution.execute(InterceptingClientHttpRequest.java:93)
	at org.springframework.http.client.InterceptingClientHttpRequest.executeInternal(InterceptingClientHttpRequest.java:77)
	at org.springframework.http.client.AbstractBufferingClientHttpRequest.executeInternal(AbstractBufferingClientHttpRequest.java:48)
	at org.springframework.http.client.AbstractClientHttpRequest.execute(AbstractClientHttpRequest.java:66)
	at org.springframework.web.client.RestTemplate.doExecute(RestTemplate.java:776)
	... 22 common frames omitted
```

So load balancer mechanism /TLS forwarding or 
maybe nginx configuration must be changed.

File: sechub-pds-examples/gosec/docker/GoSec-Ubuntu.dockerfile [hunk 2]
```diff
@@ -6,15 +6,27 @@ FROM ${BASE_IMAGE}
...
+# Folders
 ARG PDS_FOLDER="/pds"
-ARG PDS_CHECKSUM="f3101a6a2a96b39edf493249f2300c627fa01fcb7a6b96c5ae1edfbe70a31d46"
-ARG TOOL_FOLDER="/tool"
+ARG SCRIPT_FOLDER="/scripts"
+ENV TOOL_FOLDER="/tools"
+ARG WORKSPACE="/workspace"
+
+# PDS
+ENV PDS_VERSION=0.23.0
```
Comment (de-jcup) [inline sechub-pds-examples/gosec/docker/GoSec-Ubuntu.dockerfile:17]: I think we cannot use 0.23.0 but there is a need for a 0.23.1

see https://github.com/Daimler/sechub/issues/773
This is not really an issue here, but maybe a need to wait for the update.

**Review by de-jcup — CHANGES_REQUESTED** [review]
See my comments.

File: sechub-pds-examples/gosec/50-start-multiple-docker-compose.sh (hunk 1 above)
Comment (Jeeppler) [reply sechub-pds-examples/gosec/50-start-multiple-docker-compose.sh:8]: Nginx is configured to use TLSv1.3 only (see: [docker/loadbalancer/nginx.conf  -> line 22](https://github.com/Daimler/sechub/pull/696/files#diff-a2e5a11f90e2d3be3051ff5a20065c15caaaba502db17b91044db9986a6bbc00R22))

File: sechub-pds-examples/gosec/README.adoc [hunk 3]
```diff
@@ -394,66 +540,146 @@ docker exec -it pds_gosec_alpine sh
...
+pds-gosec-545f5bc8-px9cs         1/1     Running   0          1m43s
+pds-gosec-545f5bc8-t52p6         1/1     Running   0          3m
+
 ----
 
 . Forward port to own machine
 +
 ----
-kubectl port-forward pds-gosec-749fcb8d7f-jjqwn 8444:8444
+kubectl port-forward pds-gosec-545f5bc8-7s6rh  8444:8444
```
[the remark this answers is not in the document]
Comment (Jeeppler) [reply sechub-pds-examples/gosec/README.adoc:669]: @sven-dmlr what is your opinion?

[the remark this answers is not in the document]
Comment (sven-dmlr) [reply sechub-pds-examples/gosec/README.adoc:669]: You could say
`Forward port of one of the pods to own machine`
Then it should be clear to everybody.

File: sechub-pds-examples/gosec/docker/GoSec-Ubuntu.dockerfile (hunk 2 above)
Comment (Jeeppler) [reply sechub-pds-examples/gosec/docker/GoSec-Ubuntu.dockerfile:17]: I would like to merge it and later we can create a new release of the PDS and integrate it.

File: sechub-pds-examples/gosec/70-test.sh [hunk 4]
```diff
@@ -0,0 +1,96 @@
+#!/usr/bin/env bash
```
[the remark this answers is not in the document]
Comment (Jeeppler) [reply sechub-pds-examples/gosec/70-test.sh:1]: fixed

File: sechub-pds-examples/gosec/README.adoc [hunk 5]
```diff
@@ -62,9 +81,12 @@ The folder contains a `pds-api.sh` script.
 +
 ----
 export PDS_SERVER=https://localhost:8444
-export PDS_USERID=pds-dev-admin
-export PDS_APITOKEN=pds-dev-apitoken
+export PDS_USERID=admin
+export PDS_APITOKEN="pds-apitoken"
```
[the remark this answers is not in the document]
Comment (Jeeppler) [reply sechub-pds-examples/gosec/README.adoc:85]: Done

File: sechub-pds-examples/gosec/README.adoc [hunk 6]
```diff
@@ -20,36 +27,48 @@ Docker and Docker-Compose need to be installed:
...
 . https://docs.docker.com/compose/install/[Install Docker-Compose]
 
-=== Start
+=== Single Instance
+
+Start a single instance either by using a Bash wrapper script or follow the steps manually.
 
 ==== Automatic
 
 The folder contains a start script which does the <<_manually, manual>> steps for you:
```
[the remark this answers is not in the document]
Comment (Jeeppler) [reply sechub-pds-examples/gosec/README.adoc:37]: I added them were appropriate.

File: sechub-pds-examples/gosec/50-start-multiple-docker-compose.sh (hunk 1 above)
Comment (Jeeppler) [reply sechub-pds-examples/gosec/50-start-multiple-docker-compose.sh:8]: By setting `-Djavax.net.debug=all` one can get more details of the TLS connection. For more information have a look at: https://docs.oracle.com/javase/7/docs/technotes/guides/security/jsse/ReadDebug.html.

Comment (Jeeppler) [reply sechub-pds-examples/gosec/50-start-multiple-docker-compose.sh:8]: For more information about the TLS connection use the OpenSSL client: `openssl s_client -connect localhost:8444`

Comment (Jeeppler) [reply sechub-pds-examples/gosec/50-start-multiple-docker-compose.sh:8]: After enabling  `-Djavax.net.debug=all` it becomes clear, that the PDS server uses TLS 1.2 to negotiate the connection:

~~~
javax.net.ssl|FINE|4A|SecHub-exec-f40414fa-b43f-4caa-abf3-8822df8ebb0d-1|2021-09-08 20:12:18.122 CEST|ClientHello.java:575|Produced ClientHello handshake message (
"ClientHello": {
  "client version"      : "TLSv1.2",
  "random"              : "4F 48 AD C0 6C 85 7E 30 ED 0E 9B BA E9 C6 33 AE 24 64 38 DB A8 05 6B C3 03 94 29 A7 C9 6D 01 58",
  "session id"          : "",
  "cipher suites"       : "[TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384(0xC02C), TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256(0xC02B), TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384(0xC030), TLS_RSA_WITH_AES_256_GCM_SHA384(0x009D), TLS_ECDH_ECDSA_WITH_AES_256_GCM_SHA384(0xC02E), TLS_ECDH_RSA_WITH_AES_256_GCM_SHA384(0xC032), TLS_DHE_RSA_WITH_AES_256_GCM_SHA384(0x009F), TLS_DHE_DSS_WITH_AES_256_GCM_SHA384(0x00A3), TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256(0xC02F), TLS_RSA_WITH_AES_128_GCM_SHA256(0x009C), TLS_ECDH_ECDSA_WITH_AES_128_GCM_SHA256(0xC02D), TLS_ECDH_RSA_WITH_AES_128_GCM_SHA256(0xC031), TLS_DHE_RSA_WITH_AES_128_GCM_SHA256(0x009E), TLS_DHE_DSS_WITH_AES_128_GCM_SHA256(0x00A2), TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA384(0xC024), TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA384(0xC028), TLS_RSA_WITH_AES_256_CBC_SHA256(0x003D), TLS_ECDH_ECDSA_WITH_AES_256_CBC_SHA384(0xC026), TLS_ECDH_RSA_WITH_AES_256_CBC_SHA384(0xC02A), TLS_DHE_RSA_WITH_AES_256_CBC_SHA256(0x006B), TLS_DHE_DSS_WITH_AES_256_CBC_SHA256(0x006A), TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA(0xC00A), TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA(0xC014), TLS_RSA_WITH_AES_256_CBC_SHA(0x0035), TLS_ECDH_ECDSA_WITH_AES_256_CBC_SHA(0xC005), TLS_ECDH_RSA_WITH_AES_256_CBC_SHA(0xC00F), TLS_DHE_RSA_WITH_AES_256_CBC_SHA(0x0039), TLS_DHE_DSS_WITH_AES_256_CBC_SHA(0x0038), TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA256(0xC023), TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256(0xC027), TLS_RSA_WITH_AES_128_CBC_SHA256(0x003C), TLS_ECDH_ECDSA_WITH_AES_128_CBC_SHA256(0xC025), TLS_ECDH_RSA_WITH_AES_128_CBC_SHA256(0xC029), TLS_DHE_RSA_WITH_AES_128_CBC_SHA256(0x0067), TLS_DHE_DSS_WITH_AES_128_CBC_SHA256(0x0040), TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA(0xC009), TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA(0xC013), TLS_RSA_WITH_AES_128_CBC_SHA(0x002F), TLS_ECDH_ECDSA_WITH_AES_128_CBC_SHA(0xC004), TLS_ECDH_RSA_WITH_AES_128_CBC_SHA(0xC00E), TLS_DHE_RSA_WITH_AES_128_CBC_SHA(0x0033), TLS_DHE_DSS_WITH_AES_128_CBC_SHA(0x0032), TLS_EMPTY_RENEGOTIATION_INFO_SCSV(0x00FF)]",
  "compression methods" : "00",
  "extensions"          : [
    "supported_groups (10)": {
      "versions": [secp256r1, secp384r1, secp521r1, ffdhe2048, ffdhe3072, ffdhe4096, ffdhe6144, ffdhe8192]
    },
    "ec_point_formats (11)": {
      "formats": [uncompressed]
    },
    "signature_algorithms (13)": {
      "signature schemes": [ecdsa_secp256r1_sha256, ecdsa_secp384r1_sha384, ecdsa_secp521r1_sha512, rsa_pss_rsae_sha256, rsa_pss_rsae_sha384, rsa_pss_rsae_sha512, rsa_pss_pss_sha256, rsa_pss_pss_sha384, rsa_pss_pss_sha512, rsa_pkcs1_sha256, rsa_pkcs1_sha384, rsa_pkcs1_sha512, dsa_sha256, ecdsa_sha224, rsa_sha224, dsa_sha224, ecdsa_sha1, rsa_pkcs1_sha1, dsa_sha1]
    },
    "signature_algorithms_cert (50)": {
      "signature schemes": [ecdsa_secp256r1_sha256, ecdsa_secp384r1_sha384, ecdsa_secp521r1_sha512, rsa_pss_rsae_sha256, rsa_pss_rsae_sha384, rsa_pss_rsae_sha512, rsa_pss_pss_sha256, rsa_pss_pss_sha384, rsa_pss_pss_sha512, rsa_pkcs1_sha256, rsa_pkcs1_sha384, rsa_pkcs1_sha512, dsa_sha256, ecdsa_sha224, rsa_sha224, dsa_sha224, ecdsa_sha1, rsa_pkcs1_sha1, dsa_sha1]
    },
    "extended_master_secret (23)": {
      <empty>
    },
    "supported_versions (43)": {
      "versions": [TLSv1.2]
    }
  ]
}
)
~~~

However, Nginx uses TLSv1.3 only.

Comment (Jeeppler) [reply sechub-pds-examples/gosec/50-start-multiple-docker-compose.sh:8]: The solution is to add TLSv1.2 support for Ngnix.

File: sechub-pds-examples/gosec/docker/loadbalancer/nginx.conf [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+# Based on config from: https://stackoverflow.com/a/54388725
+
+worker_processes  1;
+
+events {
+  worker_connections  1024;
+}
+
+http {
+  access_log /dev/stdout;
```
Comment (sven-dmlr) [inline sechub-pds-examples/gosec/docker/loadbalancer/nginx.conf:12]: Propose to add:
```
        sendfile on;
        tcp_nopush on;
        tcp_nodelay on;
        keepalive_timeout 65;
        types_hash_max_size 2048;
```

File: sechub-pds-examples/gosec/docker/loadbalancer/nginx.conf [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+    listen 8444 ssl;
+    server_name localhost;
+
+    ssl_certificate /certificates/localhost.cert;
+    ssl_certificate_key /certificates/localhost.key;
+
+    ssl_protocols TLSv1.2 TLSv1.3;
+    ssl_prefer_server_ciphers on;
+    ssl_dhparam /certificates/certsdhparam.pem;
+    ssl_ecdh_curve secp384r1;
```
Comment (sven-dmlr) [inline sechub-pds-examples/gosec/docker/loadbalancer/nginx.conf:25]: a known good ssl setup is this:
```
        ssl_protocols TLSv1.2 TLSv1.3;
        ssl_prefer_server_ciphers on;
        ssl_ciphers  HIGH:!aNULL:!MD5:!RC4;
        ssl_session_timeout 1d;
        ssl_session_cache shared:SSL:50m;
        ssl_stapling on;
        ssl_stapling_verify on;
        add_header Strict-Transport-Security max-age=15768000;
```
But I didn't compare with your settings; maybe they overlap.

**Review by sven-dmlr — APPROVED** [review]

File: sechub-pds-examples/gosec/50-start-multiple-docker-compose.sh (hunk 1 above)
Comment (de-jcup) [reply sechub-pds-examples/gosec/50-start-multiple-docker-compose.sh:8]: **Just for documentation:**
We have inspected the reason why we need TLSv1.2 here:
When using Java 8 (even the latest one), TLSv1.3 is NOT enabled per default for client operations.
Please refer https://www.oracle.com/java/technologies/javase/8u261-relnotes.html for details.

So when starting SecHub server with Java8 the adapter will only be capable to connect via TLSv1.2 which is accepted by server side with correct ciphers and no SHA1 etc. (see https://github.com/Daimler/sechub/issues/625)

File: sechub-pds-examples/gosec/docker/loadbalancer/nginx.conf [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+
```
Comment (winzj) [inline sechub-pds-examples/gosec/docker/loadbalancer/nginx.conf:26]: I think the Diffie-Hellman configuration is good.
If it is not too restrictive, you could consider the following ciphers:
```
ssl_ciphers EECDH+AESGCM:EDH+AESGCM;
```

**Review by winzj — APPROVED** [review]

**Comment by de-jcup** [comment]
Found a HTTP 413 problem because `client_max_body_size` was not configured in nginx configuration file.

Created additional commit and changed to 20 Megabyte max upload size. PDS upload constraints are still in charge and will per default block uploads > 5MB. But loadbalancer (nginx) in our gosec-cluster variant (docker compose) will accept it. So we only must configure our "real" max size in PDS server...

File: sechub-pds-examples/gosec/docker/GoSec-Ubuntu.dockerfile (hunk 2 above)
Comment (de-jcup) [reply sechub-pds-examples/gosec/docker/GoSec-Ubuntu.dockerfile]: Just for documentaion: PDS 0.23.1 is now used.

**Review by de-jcup — APPROVED** [review]
Well done ...

**Comment by de-jcup** [comment]
@Jeeppler : I gave my approval. There are still some optional suggestions from the other reviewers - maybe you want to take a look. From my side it is okay and can be merged. It works now well with SecHub Server (develop branch) as single and also as cluster variant.

**Only for documentation:**
One problem while doing full manual integration tests with SecHub Server were Scan time outs. But this was a SecHub server bug .  I created  https://github.com/Daimler/sechub/issues/785 and fixed it already. The workaround was/is to define a dedicated timeout in minutes on executor configuration - the defaults did not make sense (check and timeout were both 1 minute...)

File: sechub-pds-examples/gosec/docker/loadbalancer/nginx.conf (hunk 9 above)
Comment (Jeeppler) [reply sechub-pds-examples/gosec/docker/loadbalancer/nginx.conf:32]: I tested the `ssl_ciphers` and it works. As a result, I included your suggestion.

File: sechub-pds-examples/gosec/docker/loadbalancer/nginx.conf (hunk 8 above)
Comment (Jeeppler) [reply sechub-pds-examples/gosec/docker/loadbalancer/nginx.conf:31]: I added some of them. However, `ssl_stapling` does not make sense for a self-signed certificate.

File: sechub-pds-examples/gosec/docker/loadbalancer/nginx.conf (hunk 7 above)
Comment (Jeeppler) [reply sechub-pds-examples/gosec/docker/loadbalancer/nginx.conf:14]: I included them.

File: sechub-pds-examples/gosec/docker/database/postgresql.conf [hunk 10]
```diff
@@ -0,0 +1,781 @@
+# -----------------------------
+# PostgreSQL configuration file
+# -----------------------------
+#
+# This file consists of lines of the form:
```
[the remark this answers is not in the document]
Comment (Jeeppler) [reply sechub-pds-examples/gosec/docker/database/postgresql.conf:5]: I did remove all unnecessary comments.

File: sechub-pds-examples/gosec/docker/database/pg_hba.conf [hunk 11]
```diff
@@ -0,0 +1,98 @@
+# PostgreSQL Client Authentication Configuration File
+# ===================================================
+#
+# Refer to the "Client Authentication" section in the PostgreSQL
+# documentation for a complete description of this file.  A short
+# synopsis follows.
```
[the remark this answers is not in the document]
Comment (Jeeppler) [reply sechub-pds-examples/gosec/docker/database/pg_hba.conf:6]: I did remove all unnecessary comments.

**Pull request merged**